tools/d2g/d2gtest/testdata/testcases/docker_image_tests.yml

```diff
@@ -62,15 +62,16 @@ testSuite:
           - - bash
             - '-cx'
             - >-
-              curl -fsSL -o path "${TASKCLUSTER_PROXY_URL}/index/v1/task/test.namespace/artifacts/test/path"
-
-              IMAGE_NAME=$(podman load -i path | sed -n '1s/.*: //p')
-
               podman run -t --rm
               -e RUN_ID -e TASKCLUSTER_ROOT_URL -e
               TASKCLUSTER_WORKER_LOCATION -e TASK_ID
-              "${IMAGE_NAME}" 'echo "Hello world"'
+              docker-archive:dockerimage 'echo "Hello world"'
```

Comment:
Is this a hardcoded name for all images? 
If worker is configured to run multiple tasks concurrently, wouldn't it cause a collision if two tasks would request two different indexed images, but the registry on the host is the same? 
Or is there some sort of isolation happening for this scenario?

Reply:
Good eye. Yes, this is hardcoded for all images. Generic worker's capacity is ever 1, though. So it doesn't and can't run tasks concurrently.

Reply:
When (if ever) https://bugzilla.mozilla.org/show_bug.cgi?id=1609102 is fixed, it will be possible to run multiple concurrent tasks under multiuser engine. However, since the `podman run` commands are executed as the task user, I suspect that the cached images and tags will be stored on a per-user basis, rather than a global cache on the host. In other words, I suspect that if OS user A tags a docker image with the name `dockerimage`, and OS user B tags a different docker image with the name `dockerimage`, that they won't conflict. But we should check this... Nice spot, @lotas!